// mirrorx/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Route", {
  enumerable: true,
  get: function get() {
    return _reactRouter.Route;
  }
});
Object.defineProperty(exports, "Redirect", {
  enumerable: true,
  get: function get() {
    return _reactRouter.Redirect;
  }
});
Object.defineProperty(exports, "Switch", {
  enumerable: true,
  get: function get() {
    return _reactRouter.Switch;
  }
});
Object.defineProperty(exports, "Prompt", {
  enumerable: true,
  get: function get() {
    return _reactRouter.Prompt;
  }
});
Object.defineProperty(exports, "withRouter", {
  enumerable: true,
  get: function get() {
    return _reactRouter.withRouter;
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _reactRouterDom.Link;
  }
});
Object.defineProperty(exports, "NavLink", {
  enumerable: true,
  get: function get() {
    return _reactRouterDom.NavLink;
  }
});
Object.defineProperty(exports, "connect", {
  enumerable: true,
  get: function get() {
    return _reactRedux.connect;
  }
});
Object.defineProperty(exports, "model", {
  enumerable: true,
  get: function get() {
    return _model.default;
  }
});
Object.defineProperty(exports, "actions", {
  enumerable: true,
  get: function get() {
    return _actions.actions;
  }
});
Object.defineProperty(exports, "render", {
  enumerable: true,
  get: function get() {
    return _render.default;
  }
});
Object.defineProperty(exports, "hook", {
  enumerable: true,
  get: function get() {
    return _hook.default;
  }
});
Object.defineProperty(exports, "Router", {
  enumerable: true,
  get: function get() {
    return _router.default;
  }
});
Object.defineProperty(exports, "defaults", {
  enumerable: true,
  get: function get() {
    return _defaults.default;
  }
});
exports.default = void 0;

var _reactRouter = require("react-router");

var _reactRouterDom = require("react-router-dom");

var _reactRedux = require("react-redux");

var _model = _interopRequireDefault(require("./model"));

var _actions = require("./actions");

var _render = _interopRequireDefault(require("./render"));

var _hook = _interopRequireDefault(require("./hook"));

var _router = _interopRequireDefault(require("./router"));

var _defaults = _interopRequireDefault(require("./defaults"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  model: _model.default,
  actions: _actions.actions,
  hook: _hook.default,
  defaults: _defaults.default,
  connect: _reactRedux.connect,
  render: _render.default,
  Router: _router.default,
  Route: _reactRouter.Route,
  Link: _reactRouterDom.Link,
  NavLink: _reactRouterDom.NavLink,
  Switch: _reactRouter.Switch,
  Redirect: _reactRouter.Redirect,
  Prompt: _reactRouter.Prompt,
  withRouter: _reactRouter.withRouter
};
exports.default = _default;
